amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTaker.java

```diff
@@ -339,16 +339,36 @@ private void updateAllLeases(Callable<Long> timeProvider)
                 return leasesToTake;
             }
 
+            int currentLeaseCount = leaseCounts.get(workerIdentifier);
+            // If there are leases that have been expired for an extended period of
+            // time, take them with priority, disregarding the target (computed
+            // later) but obeying the maximum limit per worker.
+            List<Lease> veryOldLeases = allLeases.values().stream()
+                    .filter(lease -> System.nanoTime() - lease.lastCounterIncrementNanos()
+                            > VERY_OLD_LEASE_DURATION_NANOS_MULTIPLIER * leaseDurationNanos)
+                    .collect(Collectors.toList());
+
+            if (!veryOldLeases.isEmpty()) {
+                Collections.shuffle(veryOldLeases);
+                int n = Math.max(0, Math.min(maxLeasesForWorker - currentLeaseCount, veryOldLeases.size()));
+                HashSet<Lease> result = new HashSet<>(veryOldLeases.subList(0, n));
+                if (n > 0) {
+                    log.info("Taking leases that have been expired for a long time: {}", result);
+                }
+                scope.addData("VeryOldLeases", n, StandardUnit.COUNT, MetricsLevel.SUMMARY);
```

Comment:
Can we emit rest of the metrics, as part of this?

Reply:
Also emit VeryOldLeases as part of the bottom metrics, basically move the metrics emission to a method and call from both the places.